test/C/src/no_inlining/CountHierarchy.lf

```diff
@@ -15,9 +15,9 @@ main reactor {
 
   state count: int
 
-  reaction(t.out) named increment
+  reaction increment(t.out);
 
-  reaction(t.out) named check_done
+  reaction check_done(t.out);
```

Comment:
I wonder if the semicolon should be made optional? Here there is no parsing ambiguity. In languages with optional semis, semis are only required when they resolve an ambiguity, so it feels odd that these lines have to have a semi.

I would find it more consistent to be able to write
```
state count: int
reaction increment(t.out)
reaction check_done(t.out) -> x
```
The ambiguity is only there if the following token is an identifier:
```
reaction foo(a)
a -> b
```
We don't keep track of newlines in the parser but we have access to line numbers in the validator. We could issue a warning if we find a reaction with no body, and both a source and effect declaration which are not on the same line as the reaction. Since AFAIK sources are relatively rare, I don't think this should have a lot of false positives.

Reply:
ANTLR doesn't like ambiguous grammars. Is there a way to implement what you suggest without this resulting in a barrage of errors/warnings from Xtext?

Reply:
OK, I was able to force the parse to go down the "sources" path using the `=>` predicate (see caa69f615). This removes the ambiguity, but it also means that
```
reaction foo(x)
p -> q

```
will be parsed as:
```
reaction foo(x) p -> q
```
For that particular situation, we need to add a validator check to tell the user that it must place a `;` after the reaction declaration.